src/main/java/teammates/storage/search/InstructorSearchDocument.java

```diff
@@ -76,12 +72,10 @@ public static InstructorSearchResultBundle fromResults(Results<ScoredDocument> r
         }
 
         for (ScoredDocument doc : results) {
-            InstructorAttributes instructor = JsonUtils.fromJson(
-                    doc.getOnlyField(Const.SearchDocumentField.INSTRUCTOR_ATTRIBUTE).getText(),
-                    InstructorAttributes.class);
-
-            if (instructorsDb.getInstructorForRegistrationKey(StringHelper.encrypt(instructor.key)) == null) {
-                instructorsDb.deleteDocument(instructor);
+            InstructorAttributes instructor = instructorsDb.getInstructorForRegistrationKey(doc.getId());
+            if (instructor == null) {
+                // search engine out of sync
```

Comment:
I think you should copy the complete version of the comment here (as what you did in FRCSearchDocument) as well.